### PR TITLE
Universal Object Damage

### PR DESCRIPTION
**Pull request by Fox-McCloud**
This needs to be done. Need I say more?

Obviously WIP and obviously DNM.

Universal damage turned on; acid melts things over time now; armor values now have acid and fire attached to them.

Should fix a ton of things including lava's behavior--also means a mostly destructable environment.

img
img

fixes: https://github.com/ParadiseSS13/Paradise/issues/12246
fixes: https://github.com/ParadiseSS13/Paradise/issues/11902
fixes: https://github.com/ParadiseSS13/Paradise/issues/10930
fixes: https://github.com/ParadiseSS13/Paradise/issues/2805
fixes: https://github.com/ParadiseSS13/Paradise/issues/3327

:cl: Fox McCloud
add: Most things are universally damageable
tweak: Acid melts things a bit differently now
fix: Lava actually sets most things on fire.
tweak: Captain no longer has access to his display case by default
add: Xenos can break open vents
/:cl:

### CODE REVIEW

**Comment by Aurorablade** [comment]
I like the orignal title..the theme song of this PR is now Ring of FIre

**Comment by Ty-Omaha** [comment]
can we like

shoot the tesla to kill it

or throw rubber gloves at it

**Comment by Fox-McCloud** [comment]
Also, there is probably balance altering armor changes in here for a number of things. I'm not going to document them all; for the most part, it's minor buffs/nerfs, with the exception of maybe windows. 

This is mostly copy-pasting form TG to ensure consistency in behavior--this isn't me trying to sneak in changes; it's just that with something this massive, I really don't have time/willpower to document every single armor change.

**Comment by Fox-McCloud** [comment]
There's one thing I'm not going to be porting over to this system, because they're atrocious, snowflakey, and they can get wrecked for all I care; spacepods.

The changes here shouldn't be a problem for spacepods; they *should* still behave as before, but I'm not going to invest a lot of time into getting them to work with this system.

To be blunt, spacepods should probably just be a type of mech subtype that can process space movement with little delay, but that's for another day.

File: code/ATMOSPHERICS/atmospherics.dm [hunk 1]
```diff
@@ -31,8 +31,8 @@ GLOBAL_DATUM_INIT(pipe_icon_manager, /datum/pipe_icon_manager, new())
 	var/image/pipe_image
 
 /obj/machinery/atmospherics/New()
-	if(!armor)
-		armor = list(melee = 25, bullet = 10, laser = 10, energy = 100, bomb = 0, bio = 100, rad = 100)
+	if (!armor)
```
Comment (Couls) [inline code/ATMOSPHERICS/atmospherics.dm]: space

**Comment by Fox-McCloud** [comment]
My face when someone set `unacidable` all over the place on an away mission.

No wonder TG considers those things almost universally trash.

**Comment by KasparoVy** [comment]
Polyacid your way to victory comrade

**Comment by Fox-McCloud** [comment]
had to do a couple of refactors in the way; display cases and ash trays. 

Can't construct alarmed display cases, the cap's display case is no longer fingerprint locked, and it has access that not even cap has (he'll have to break it).

I may change the latter most part of it, but we'll see; it was mostly just trying to keep it up to date and avoiding the problems that the display case would cause with how object damage works.

Either case, most `var/health` is killed off. 

Shields and shield generators still use it, because they need an extensive refactor to work with this system--portable turrets also have `var/health` I'm not sure if I'll tweak them, revert them back to pre universal object damage....or what; the three problem things with this PR are space pods, portable turrets, and field emitters/forcefield generators/shields.

Either case, I'll think about what to do with portable turrets; once I've decided and handled that (or not), then it's onto the final stretch; `attackby` returning parent value.

Then, and finally then, I can start testing and how much this broke and how badly this works.

**Comment by Fox-McCloud** [comment]
Ok, yeah, I'm so sick of this PR and the other refactors I have to do to get it to work....

I'm going to just save the portable turrets for later, along with the field generator and space pods.

Those are all inconsequential enough (and/or work fine with their snowflake BS), this is so long overdue, and I'm so tired of working on this, that it can wait. 

In any event, this is the PR "done" from what I believe needs to be changed.

Tomorrow I start testing.

**Comment by Fox-McCloud** [comment]
Ok, another thing that will use the old system: disposals.

Disposal code is goodness *awful* and is something else that needs to be refactored.

**Comment by Fox-McCloud** [comment]
This is the best I can do. Done as much testing as I feel is necessary and within reasonable limits.

**Comment by KasparoVy** [comment]
Legendary, way to go

**Pull request merged**